stl/inc/algorithm

```diff
@@ -6628,6 +6628,154 @@ void sort(_ExPo&& _Exec, const _RanIt _First, const _RanIt _Last) noexcept /* te
+            // swap middle and last, then test first again
+            _RANGES iter_swap(_Last, _Mid);
+
+            if (_STD invoke(_Pred, _STD invoke(_Proj, *_Mid), _STD invoke(_Proj, *_First))) {
+                _RANGES iter_swap(_Mid, _First);
+            }
+        }
+    }
+
+    template <class _It, class _Pr, class _Pj>
+    constexpr void _Guess_median_unchecked(_It _First, _It _Mid, _It _Last, _Pr _Pred, _Pj _Proj) {
+        // sort median element to middle
+        _STL_INTERNAL_STATIC_ASSERT(random_access_iterator<_It>);
+        _STL_INTERNAL_STATIC_ASSERT(sortable<_It, _Pr, _Pj>);
+
+        using _Diff        = iter_difference_t<_It>;
+        const _Diff _Count = _Last - _First;
+        if (_Count > 40) { // Tukey's ninther
+            const _Diff _Step     = (_Count + 1) >> 3; // +1 can't overflow because range was made inclusive in caller
+            const _Diff _Two_step = _Step << 1; // note: intentionally discards low-order bit
+            _Med3_unchecked(_First, _First + _Step, _First + _Two_step, _Pred, _Proj);
+            _Med3_unchecked(_Mid - _Step, _Mid, _Mid + _Step, _Pred, _Proj);
+            _Med3_unchecked(_Last - _Two_step, _Last - _Step, _Last, _Pred, _Proj);
+            _Med3_unchecked(_First + _Step, _Mid, _Last - _Step, _Pred, _Proj);
+        } else {
+            _Med3_unchecked(_First, _Mid, _Last, _Pred, _Proj);
+        }
+    }
+
+    template <class _It, class _Pr, class _Pj>
+    constexpr subrange<_It> _Partition_by_median_guess_unchecked(_It _First, _It _Last, _Pr _Pred, _Pj _Proj) {
+        // Choose a pivot, partition [_First, _Last) into elements less than pivot, elements equal to pivot, and
+        // elements greater than pivot; return the equal partition as a subrange.
+        _STL_INTERNAL_STATIC_ASSERT(random_access_iterator<_It>);
+        _STL_INTERNAL_STATIC_ASSERT(sortable<_It, _Pr, _Pj>);
+
+        _It _Mid = _First + ((_Last - _First) >> 1); // shift for codegen
+        _RANGES _Guess_median_unchecked(_First, _Mid, _RANGES prev(_Last), _Pred, _Proj);
+        _It _Pfirst = _Mid;
+        _It _Plast  = _RANGES next(_Pfirst);
+
+        while (_First < _Pfirst
+               && !_STD invoke(_Pred, _STD invoke(_Proj, *_RANGES prev(_Pfirst)), _STD invoke(_Proj, *_Pfirst))
+               && !_STD invoke(_Pred, _STD invoke(_Proj, *_Pfirst), _STD invoke(_Proj, *_RANGES prev(_Pfirst)))) {
+            --_Pfirst;
+        }
+
+        while (_Plast < _Last && !_STD invoke(_Pred, _STD invoke(_Proj, *_Plast), _STD invoke(_Proj, *_Pfirst))
+               && !_STD invoke(_Pred, _STD invoke(_Proj, *_Pfirst), _STD invoke(_Proj, *_Plast))) {
+            ++_Plast;
+        }
+
+        _It _Gfirst = _Plast;
+        _It _Glast  = _Pfirst;
+
+        for (;;) { // partition
+            for (; _Gfirst < _Last; ++_Gfirst) {
+                if (_STD invoke(_Pred, _STD invoke(_Proj, *_Pfirst), _STD invoke(_Proj, *_Gfirst))) {

```

Comment:
Add a `continue` for readability?

Reply:
Added in both places, and in both places in `std::_Partition_by_median_guess_unchecked`.